### DISCUSSION TITLE
NU HTML Checker says: An input element with a type attribute whose value is hidden must not have an autocomplete attribute whose value is on or off.

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#1507] [PrimeFaces] [answered]
Nu THML Checker says, that using autocomplete in a hidden input element is not valid: 
![image](https://github.com/primefaces/community/assets/33621098/1ec59927-e113-4548-bc82-e19bb199f67d)

Can you please check this? Thanks

**Comment by melloware** [comment]
First that ViewState is coming from Mojarra or MyFaces as that is the JSF library not PrimeFaces.

Second, I wouldn't trust NuHTML checker for stuff like this....only trust a real screen reader.  PrimeFaces has always set AutoComplete="off" on hidden inputs because a lot of data is passed to the backend through hidden inputs and we don't want the browser rememebering those values.

From ChatGPT:

>  Using autocomplete="off" can indeed be useful to prevent browsers from autofilling input fields. However, there are some considerations to keep in mind:
> 
> Hidden inputs: If you're using type="hidden", it's generally not visible to users, so autofilling it might not be a significant concern. However, there could be cases where sensitive data is passed through hidden inputs, and in those cases, autocomplete="off" could be useful.
> 
> In conclusion, using autocomplete="off" with type="hidden" can be appropriate in certain situations, particularly if you're dealing with sensitive data. However, be aware of its limitations and potential impact on user experience and security.

**Reply by melloware** [reply, marked as answer]
OK I opened this Faces Spec question/issue: https://github.com/jakartaee/faces/issues/1912

**Comment by hgfrey** [comment]
The NU-HTML-Checker is not longer relevant. Here is a message from our accessibility team:

WCAG 2.1 update
The success criterion 4.1.1 Correct syntax was adjusted in an update to WCAG 2.1. According to the adaptation, the success criterion is always considered to be met if the content is displayed in HTML or XML.

The adjustment was made because the HTML standard has changed since the test step was created and there are now standardized procedures for user agents (browsers) to deal with, for example, unclosed HTML tags. However, since syntactic problems are also addressed by other test criteria and this test step also reveals problems that are not relevant to accessibility, the ITZBund has decided to update the test criterion.

The test step must therefore always be set to fulfilled.

**Reply by melloware** [reply]
@hgfrey there is also a setting in Mojarra to remove that autocomplete value if you don't want the NuHTml checker reporting it: https://stackoverflow.com/questions/18460909/firefox-overrides-javax-faces-viewstate-with-old-value-on-f5-how-to-turn-off-au
